test/e2e/cypress/e2e/settings.cy.js

```diff
@@ -677,4 +677,133 @@ context('Settings page', () => {
       });
     });
   });
+
+  describe('Activity log', () => {
+    // Elements must be filtered to match only the ones relate to the Activity Logs.
+    // Readable texts is contained by the section,
+    // while inputs are contained by the modal.
+    // The use of helpers instead of Cypress alias avoid problems with the React rendering,
+    // as the alias is not updated when the component is re-rendered.
+    const section = () => cy.contains('Activity Logs').parents('section');
+    const modal = () =>
+      cy.contains('Enter Activity Logs Settings').parents('div').first();
+
+    before(() => {
+      cy.reload();
+    });
+
+    beforeEach(() => {
+      // Read the initial retention time to compare with the new value
+      // Also, it checks the retention time is displayed correctly
+      section()
+        .get('[aria-label="retention-time"]')
+        .invoke('text')
+        .as('initialRetentionTime');
+    });
+
+    describe.only('Changing Settings', () => {
+      it('should change retention time', () => {
+        // the initial retention time is read before the settings are changed
+        cy.get('@initialRetentionTime').then((text) =>
+          section()
+            .get('[aria-label="retention-time"]')
+            .should('have.text', text)
```

Comment:
how does this change from what we do in `beforeEach`? Are they both needed?

Reply:
In the `beforeEach` clause we read the initial text, here we ensure the test starts from there.

It's indeed a step that never fails, I mostly added it for completing the test case (start from a value x, get another value y). Should we remove it? I see no harm, I wonder if it can work as a "safety net" for regressions.